### ISSUE TITLE
Create ability for professionals to select the solutions they offer

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#807]
## To-do
- [ ] Create UserListingSolution table
Table stores the solutions the user offers for each listing.
- [ ] Create JobTitleSolution 
Map solutions for each job title
- [ ] Show suggested solutions for the given job title
- [ ] Let users choose solutions that were not suggested
Create component that searches all solutions and adds them to UserListingSolution

**Comment by IagoSRL** [comment]
FK for UserListingSolution added as of #812.

**Comment by joshdanielson** [comment]
@IagoSRL let's determine how straight-forward this is as it could be a lower priority if more than a half-day's work.

**Comment by IagoSRL** [comment]
Hi @joshdanielson this is **done** and uploaded to *dev* (and branch) for testing. I suppose all existent DB scripts were already executed there so I just run last ones I created this days.

Notes:
- Both public listing and listingEditor shows the list of user solutions
- 'defaultSelected' JobTitleSolutions are added on listing creation
- An upgrade process is applied so existent listings get the defaultSelected applied.
- The listing-solutions-editor is complete, with suggestions and autocomplete
- The visual appearance of solutions list can get polished but not a priority.
- Some DB problems: I had to delete the next list of records from JobTitleSolution, because they had a solutionID that does NOT exist at [solution] table (throwing errors when applying the 'defaultSelected); I added a FK to the table to prevent that from happening again. Here is the list:

```
JobTitleID	SolutionID	LanguageID	CountryID	DefaultSelected	DisplayRank	CreatedDate	UpdatedDate	ModifiedBy
36	260	1	1	0	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
37	260	1	1	1	40	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
37	262	1	1	1	30	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
72	264	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
90	270	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
92	263	1	1	1	100	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
128	267	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
155	258	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
156	258	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
179	262	1	1	0	60	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
198	270	1	1	0	70	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
248	262	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
1143	265	1	1	0	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
1198	259	1	1	1	90	2018-03-05 11:05:00.0000000 +00:00	2018-03-05 00:00:00.0000000 +00:00	jd
```

Please test and report. ASAP we release this the better.

**Comment by joshdanielson** [comment]
@iagosrl tested a few of my listings and all good!

**Comment by IagoSRL** [comment]
OK, thanks. Then, I will merge this epic branch to master and from there to is814-EPIC-partner-metrics branch. I will close this branch and issue; I see there is an issue for Phase-1 on this epic still, but I think make more sense to close this epic branches once the current dev phase is done; it's confusing to see the branches still open and conflict possibilities increase; any further phase issue can have it's own new branch created from master when is started. Sounds good?

**Comment by joshdanielson** [comment]
@iagosrl yes, close the epics and leave the few issues left open.
